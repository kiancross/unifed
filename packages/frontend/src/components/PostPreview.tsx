import React from "react";
import { Card, CardActionArea, CardContent, Grid, Link, Typography } from "@material-ui/core";
import UserIcon from "./UserIcon";
import "fontsource-roboto";

interface PostValues {
  username: string;
  title: string;
  postId: string;
}

const PostPreview = (props: PostValues): JSX.Element => {
  return (
    <Grid item container spacing={2}>
      <Grid item xs={1} container justify="flex-end">
        <UserIcon username={props.username} small />
      </Grid>
      <Grid item xs={11} container direction="column" justify="flex-start">
        <Card style={{ textAlign: "left" }}>
          <CardActionArea disableRipple href={"/posts/" + props.postId}>
            <CardContent>
              <Typography variant="body2" gutterBottom>
                <Link href={"/user/" + props.username}>{props.username}</Link>
              </Typography>
              <Typography variant="body1">{props.title}</Typography>
            </CardContent>
          </CardActionArea>
        </Card>
      </Grid>
    </Grid>
  );
};

export default PostPreview;
