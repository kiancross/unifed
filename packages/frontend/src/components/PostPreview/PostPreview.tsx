/*
 * CS3099 Group A3
 */

import { ReactElement, useState } from "react";
import { Link as RouterLink } from "react-router-dom";
import { Card, CardActionArea, CardContent, Grid, Typography, makeStyles } from "@material-ui/core";
import { PostEditor, PostHeader } from "..";

interface PostValues {
  username: string;
  title: string;
  id: string;
  server: string;
  community: string;
  body: string;
}

const useStyles = makeStyles((theme) => ({
  card: {
    textAlign: "left",
    background: theme.palette.secondary.main,
  },
}));

export function PostPreview(props: PostValues): ReactElement {
  const [editorOpen, setEditorOpen] = useState(false);
  const classes = useStyles();

  const content = editorOpen ? (
    <PostEditor
      body={props.body}
      title={props.title}
      server={props.server}
      id={props.id}
      submitButtonText="Save Post"
      onSuccess={() => window.location.assign(window.location.href)}
      onCancel={() => setEditorOpen(false)}
    />
  ) : (
    <Grid item container spacing={2}>
      <Grid item container direction="column" justify="flex-start">
        <Card className={classes.card}>
          <PostHeader
            community={props.community}
            onToggleEdit={() => setEditorOpen(true)}
            title={props.title}
            username={props.username}
            id={props.id}
            server={props.server}
            isPreview
          />
          <CardActionArea
            disableRipple
            component={RouterLink}
            to={`/instances/${props.server}/communities/${props.community}/posts/${props.id}`}
          >
            <CardContent>
              <Typography variant="body1">{props.title}</Typography>
            </CardContent>
          </CardActionArea>
        </Card>
      </Grid>
    </Grid>
  );

  return content;
}
