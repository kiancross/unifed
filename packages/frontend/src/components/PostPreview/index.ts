/*
 * CS3099 Group A3
 */

export * from "./PostPreview";
