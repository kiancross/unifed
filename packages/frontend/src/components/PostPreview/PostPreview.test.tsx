import React from "react";
import PostPreview from "./PostPreview";
import renderer from "react-test-renderer";

test("Render", () => {
  const component = renderer.create(
    <PostPreview
      username="Test Name"
      title="Test Title"
      postId="1"
      server="Test Server"
      community="Test Community"
    />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
