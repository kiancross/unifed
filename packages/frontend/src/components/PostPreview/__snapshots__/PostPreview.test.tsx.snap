// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      style={
        Object {
          "height": "28px",
          "width": "28px",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        T
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
  >
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiCardActionArea-root"
        href="/instances/Test Server/communities/Test Community/posts/1"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <div
          className="MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/user/Test Name"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Test Name
            </a>
          </p>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Test Title
          </p>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </a>
    </div>
  </div>
</div>
`;
